assets/js/googlesitekit/datastore/user/date-range.js

```diff
@@ -79,6 +84,50 @@ export const selectors = {
 		const { dateRange } = state;
 		return dateRange;
 	},
+	/**
+	 * Returns the current date range as a list of date strings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object}  state                The current data store's state.
+	 * @param {Object} 	options              Options parameter. Default is: {}.
+	 * @param {number} 	options.offsetDays   Number of days to offset. Default is: 0.
+	 * @param {boolean} options.compare      Set to true if date ranges to compare should be included. Default is: false.
+	 * @param {boolean} options.weekDayAlign Set to true if the compared date range should be aligned for the weekdays. Default is: false.
+	 * @return {Array.<string>}              List of date ranges in chronoligical order. [compareStart, compareEnd, start, end, today]
+	 */
+	getDateRangeDates( state, {
+		offsetDays = 1,
+		compare = false,
+		weekDayAlign = false,
+	} = {} ) {
+		const dateRange = selectors.getDateRange( state );
+		const today = getDateString( new Date() );
+		const endDate = getPreviousDate( today, offsetDays );
+		const matches = dateRange.match( '-(.*)-' );
+		const numberOfDays = Number( matches ? matches[ 1 ] - 1 : 28 );
+		const startDate = getPreviousDate( endDate, numberOfDays );
+		const parsed = [];
+
+		if ( compare && ! weekDayAlign ) {
+			const compareEndDate = getPreviousDate( startDate, 1 );
+			parsed.push( getPreviousDate( compareEndDate, numberOfDays ) );
+			parsed.push( compareEndDate );
+		}
+
+		if ( weekDayAlign ) {
+			const compareEndDate = getPreviousWeekDate( endDate, numberOfDays );
+			parsed.push( getPreviousDate( compareEndDate, numberOfDays ) );
+			parsed.push( compareEndDate );
+		}
```

Comment:
I think this should be simplified:
* This should be wrapped in a single `if ( compare )` clause, because `weekdayAlign` is irrelevant otherwise.
* The assignment for `compareEndDate` is the only thing which is dependent on whether `weekdayAlign` or not.
* Let's assign these values to variables (i.e. add `compareStartDate`), no need for an array here (see other comment below).